### ISSUE TITLE
SSATags linter suggests SSA markers on field that already has listType on the type

### ISSUE THREAD

**Issue by sbueringer** [kubernetes-sigs/kube-api-linter#119]
Finding

```
api/core/v1beta2/machine_types.go:534:2: ssatags: Addresses should have a listType marker for proper Server-Side Apply behavior (atomic, set, or map) (kubeapilinter)
        Addresses MachineAddresses `json:"addresses,omitempty"`
        ^
```

Types

```go
	// addresses is a list of addresses assigned to the machine.
	// This field is copied from the infrastructure provider reference.
	// +optional
	Addresses MachineAddresses `json:"addresses,omitempty"`
```

```go
// MachineAddresses is a slice of MachineAddress items to be used by infrastructure providers.
// +kubebuilder:validation:MaxItems=32
// +listType=atomic
type MachineAddresses []MachineAddress
```

controller-gen forces me to put the listType marker on the type, KAL doesn't detect it there

**Comment by sbueringer** [comment]
@JoelSpeed If you take this PR (https://github.com/kubernetes-sigs/cluster-api/pull/12470) and remove the exclude you'll see this still happens
